homeassistant/components/risco/config_flow.py

```diff
@@ -83,11 +85,26 @@ async def validate_local_input(
 
     Data has the keys from LOCAL_SCHEMA with values provided by the user.
     """
-    risco = RiscoLocal(data[CONF_HOST], data[CONF_PORT], data[CONF_PIN])
-    await risco.connect()
+    attempts = 0
+
+    while True:
+        try:
+            risco = RiscoLocal(
+                data[CONF_HOST],
+                data[CONF_PORT],
+                data[CONF_PIN],
+                **{CONF_COMMUNICATION_DELAY: attempts},
+            )
+            await risco.connect()
+            break
+        except CannotConnectError as e:
+            if attempts >= MAX_COMMUNICATION_DELAY:
+                raise e
+            attempts += 1
+
     site_id = risco.id
     await risco.disconnect()
-    return {"title": site_id}
+    return {"title": site_id, "attempts": str(attempts)}
```

Comment:
Why are we converting to string here? Keep it as an integer if that's what we want to use.